core/state/state_object.go

```diff
@@ -351,6 +361,13 @@ func (s *stateObject) updateTrie(db Database) Trie {
 		}
 		usedStorage = append(usedStorage, common.CopyBytes(key[:])) // Copy needed for closure
 	}
+	if len(keysToDelete) > 0 {
+		sort.Slice(keysToDelete, func(i, j int) bool { return keysToDelete[i].big.Cmp(keysToDelete[j].big) < 0 })
```

Comment:
I think we can directly compare `common.Hash`s as they're a `[32]byte` right? It's worth a try to simplify this I think.